### DISCUSSION TITLE
Improvements for "Sse subscription client"

### DISCUSSION THREAD

**Discussion by TheMeinerLP** [Netflix/dgs-framework#652] [Ideas]
I saw for my company we required a subscriptions for a customers api. But the current [pull request](https://github.com/Netflix/dgs-framework/pull/643) with simple subscribe was not enough. Of the reason the subscription need be written as a string. Thats looks in Java and Kotlin very unbeautiful. 
My changes what i want to do. Add code generations like queries and projections. 

Also when i have time over, i extend the documentation. 

All that inspired me was the PR of  @paulbakker 

In advance, sorry for the bad english!

**Comment by srinivasankavitha** [comment]
@TheMeinerLP - We do have code generation for subscription queries and projections as part of the dgs codegen plugin. Please try that out and let us know what is missing before adding to it.

**Reply by paulbakker** [reply]
Just yesterday I added an integration test example to one of our example apps: https://github.com/Netflix/dgs-examples-java/blob/main/src/test/java/com/example/demo/ReviewSubscriptionIntegrationTest.java.
This is using the websocket version, but it works the same.

I'm working on docs, they should be published soon too!
